### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by ALI-KHALED-13**

### CODE REVIEW

File: api/feedback/controllers.js [hunk 1]
```diff
@@ -48,36 +51,51 @@ const addFeedback = async (req, res) => {
...
+  if (  req.body.tags && 
+      targetFeedback.tags.join(",") !== req.body.tags.join(",")
+  ){
+    req.body.tags = req.body.tags.map(tag=> tag.toLowerCase());
+
+    for (let tagName of req.body.tags){
+      const tagInDB = await Tag.exists({name: tagName})
+      if (tagInDB === null){ // it's a newly added tag, add it to tags collection 
+        const tag = await Tag.create({name: tagName})
+        tag.save();
```
Comment (Elena-Erganyan) [inline api/feedback/controllers.js:68]: Add `await` please

File: api/feedback/controllers.js [hunk 2]
```diff
@@ -48,36 +51,51 @@ const addFeedback = async (req, res) => {
...
+  const feedback = await Feedback.findById(id);
+  const owner = await User.findById(req.body.owner);
+  const targetProduct = await Product.findById(req.body.targetProduct);
+  
+  if (!owner || !feedback) {
+    throw new DB404Error((owner? "feedback":"owner") +" id is missing or not found")
+  } 
+  if (targetProduct){ // only excute this block if the product is still present (as it might be deleted at one point and we are keeping the feedback)
+    targetProduct.feedback = targetProduct.feedback.filter(feedbackId=> feedbackId !== id);
+    targetProduct.save();
```
Comment (Elena-Erganyan) [inline api/feedback/controllers.js:91]: Await 😭

File: api/feedback/controllers.js [hunk 3]
```diff
@@ -48,36 +51,51 @@ const addFeedback = async (req, res) => {
...
+  }
+
+  owner.feedbackHistory = owner.feedbackHistory.filter(feedbackId=> feedbackId !== id);
+  owner.save();
```
Comment (Elena-Erganyan) [inline api/feedback/controllers.js:95]: Await

File: api/feedback/controllers.js [hunk 4]
```diff
@@ -48,36 +51,51 @@ const addFeedback = async (req, res) => {
...
+
+  await feedback.remove();
+  res.status(200).json(`feedback was updated succefully`)
```
Comment (Elena-Erganyan) [inline api/feedback/controllers.js:98]: Change the comment to "deleted successfully" and check the status code

**Review by Elena-Erganyan — CHANGES_REQUESTED** [review]

File: server.js [hunk 5]
```diff
@@ -35,7 +36,9 @@ app.use("/api/feedback", feedbacksRoutes);
 
 app.use("/api/users", usersRoutes);
 
-app.use("/api/tags", tagsRoutes)
+app.use("/api/tags", tagsRoutes);
+
+app.use("/api/status", statusRoutes)
```
Comment (Elena-Erganyan) [inline server.js]: Better to use "statuses" maybe? For consistency :) 
Not a big deal though, we can keep it this way

**Review by Elena-Erganyan — CHANGES_REQUESTED** [review]
First of all, great job 👏
Very structured and neat work.
--------
I have left some comments here and there, please have a look :)

General comment to all controllers, where the `id` is used (get one item, modify, delete): 
If the `id` is not valid the error like this one will be thrown: `internal server Error: Cast to ObjectId failed for value \"640f363970dbd0a7d994710\" (type string) at path \"_id\" for model \"feedback\` (you can see it if you pass an id which is shorter than an usual one for example).
So for the sake of error handling probably it's better to check the `id` using `mongoose.Types.ObjectId.isValid(id)` and throw some custom error.

File: api/comments/controllers.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+const addComment = async(req, res)=> {
+  const user = req.user;
+  req.body.owner = user._id;
+
+  const parentFeedback = await Feedback.findById(req.body.parentFeedback);
+
+  const comment = await Comment.create(req.body);
+
+  if (parentFeedback) { // only if it exists (it might have been deleted)
```
[the remark this answers is not in the document]
Comment (ALI-KHALED-13) [reply api/comments/controllers.js:27]: added explicitly by the script that submits the comment made

File: api/users/controllers/updateUser.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+
+const updateUser = async (req, res)=> {
+  const id = req.params.id;
+
+  const user = req.user;
+  if (user.id !== id){
+    throw new UnauthorizedError("you can edit your own info only")
+  }
+
```
[the remark this answers is not in the document]
Comment (ALI-KHALED-13) [reply api/users/controllers/updateUser.js:12]: this can be a future change where we define access more tightly, but for now the current logic works

**Review by Elena-Erganyan — APPROVED** [review]

**Pull request merged**